lib/matplotlib/axes/_axes.py

```diff
@@ -7907,8 +7907,8 @@ def violinplot(self, dataset, positions=None, vert=True, widths=0.5,
         """
 
         def _kde_method(X, coords):
-            if hasattr(X, 'values'):  # support pandas.Series
-                X = X.values
+            if hasattr(X, 'to_numpy'):  # support pandas.Series
+                X = X.to_numpy()
```

Comment:
Are there any libraries we need to worry about that implement `X.values`, but not `X.to_numpy()`? Maybe we also want to leave an `elif` with the `X.values` block as a fallback just in case...?